From 6204ce0df320b42df80284846ba3447906e4cac3 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Sat, 15 Nov 2014 23:19:45 +0000
Subject: [PATCH 2/3] ThemeInfo: add an optional 'basetheme' element to the
 theme info file

This element should be the name of the theme the theme inherits from if
applicable.
---
 mythtv/libs/libmythui/themeinfo.cpp |    5 +++++
 mythtv/libs/libmythui/themeinfo.h   |    2 ++
 2 files changed, 7 insertions(+)

diff --git a/mythtv/libs/libmythui/themeinfo.cpp b/mythtv/libs/libmythui/themeinfo.cpp
index 02f45a5..bfc3ebd 100644
--- a/mythtv/libs/libmythui/themeinfo.cpp
+++ b/mythtv/libs/libmythui/themeinfo.cpp
@@ -99,6 +99,10 @@ bool ThemeInfo::parseThemeInfo()
             {
                 m_name = getFirstText(e);
             }
+            else if (e.tagName() == "basetheme")
+            {
+                m_baseTheme = getFirstText(e);
+            }
             else if (e.tagName() == "aspect")
             {
                 m_aspect = getFirstText(e);
@@ -243,6 +247,7 @@ void ThemeInfo::ToMap(InfoMap &infoMap) const
 {
     infoMap["description"] = m_description;
     infoMap["name"] = m_name;
+    infoMap["basetheme"] = m_baseTheme;
     infoMap["aspect"] = m_aspect;
     infoMap["resolution"] = QString("%1x%2").arg(m_baseres.width())
                                             .arg(m_baseres.height());
diff --git a/mythtv/libs/libmythui/themeinfo.h b/mythtv/libs/libmythui/themeinfo.h
index 50673f4..1b8edcd 100644
--- a/mythtv/libs/libmythui/themeinfo.h
+++ b/mythtv/libs/libmythui/themeinfo.h
@@ -28,6 +28,7 @@ class MUI_PUBLIC ThemeInfo : public XMLParseBase
     QString GetAspect() const { return m_aspect; }
     const QSize   *GetBaseRes() const { return &m_baseres; }
     QString GetName() const { return m_name; }
+    QString GetBaseTheme() const { return m_baseTheme; }
     QString GetDescription() const { return m_description; }
     QString GetErrata() const { return m_errata; }
     QString GetPreviewPath() const { return m_previewpath; }
@@ -48,6 +49,7 @@ class MUI_PUBLIC ThemeInfo : public XMLParseBase
 
     QString   m_themeurl;
     QFileInfo m_theme;
+    QString   m_baseTheme;
     int       m_type;
     QString   m_aspect;
     QSize     m_baseres;
-- 
1.7.10.2

